test/es6/ES6ArrayAPI.js

```diff
@@ -904,8 +904,9 @@ var tests = [
             function test(arr, desc) {
                 desc = desc + " has species which gives an array with non-config property, validating on ";
                 var error = "Cannot redefine property '1'";
+                var errorBuiltIn = "Assignment to read-only properties is not allowed in strict mode"
                 assert.throws(function () { Array.prototype.map.call(arr, function (a) { return a;}); }, TypeError, desc + "map", error);
-                assert.throws(function () { Array.prototype.filter.call(arr, function (a) { return true;}); }, TypeError,desc + "filter", error);
+                assert.throws(function () { Array.prototype.filter.call(arr, function (a) { return true;}); }, TypeError,desc + "filter", errorBuiltIn);

```

Comment:
this shouldn't change, no? This may be happening due to CreateDataPropertyOrThrow in not there?